src/query/storages/fuse/src/io/read/block/block_reader_native.rs

```diff
@@ -163,6 +166,55 @@ impl BlockReader {
         Ok((index, fuse_reader))
     }
 
+    pub fn fill_missing_native_column_values(
+        &self,
+        data_block: DataBlock,
+        parts: &VecDeque<PartInfoPtr>,
+    ) -> Result<DataBlock> {
+        let part = FusePartInfo::from_part(&parts[0])?;
+        let column_nodes = self.projection.project_column_nodes(&self.column_nodes)?;
+        let mut need_to_fill_data = false;
+
+        let columns_meta = &part.columns_meta;
+        // check if need to fill default data
+        for column_node in column_nodes {
+            for column_id in &column_node.leaf_column_ids {
+                if !columns_meta.contains_key(column_id) {
+                    need_to_fill_data = true;
+                    break;
+                }
+            }
+            if need_to_fill_data {
+                break;
+            }
+        }
+
+        if need_to_fill_data {
+            let data_block_column_ids: HashSet<ColumnId> =
+                part.columns_meta.keys().cloned().collect();
+            let mut num_rows = 0;
+            for part in parts {
+                let part = FusePartInfo::from_part(part)?;
+                num_rows += part.nums_rows;
+            }
+
+            let data_schema = self.data_schema();
+            let ctx = self.ctx.clone();
+            let table = self.table.clone();
+            let default_vals = Self::schema_default_vals(data_schema, table, ctx)?;
```

Comment:
This method must be initialized only once, it's a very cost which calls a heavy block_on!